pkgs/by-name/ly/lyra-cursors/package.nix

```diff
@@ -0,0 +1,97 @@
+{ lib
+, stdenvNoCC
+, inkscape
+, xcursorgen
+, fetchFromGitHub
+, fetchurl
+, styles ? [ ]
+}:
+let
+  validStyles = [
+    "LyraB"
+    "LyraF"
+    "LyraG"
+    "LyraP"
+    "LyraQ"
+    "LyraR"
+    "LyraS"
+    "LyraX"
+    "LyraY"
+  ];
+
+  selectedStyles =
+    if (styles == [ ]) then
+      validStyles
+    else
+      let unknown = lib.subtractLists validStyles styles; in
+      if (unknown != [ ]) then
+        throw "Unknown style(s): ${lib.concatStringsSep " " unknown}"
+      else styles;
```

Comment:
Is the complexity of the customization point here worthwhile? The AUR package in the fork simply builds all these styles, which seems like a simpler option to me. Hydra will handle doing the build for users, anyway, so it would make things slower for most users to customize this.

Reply:
Since the build script already expects the names of the styles to build, I don't think this is a lot of extra effort. I don't see a reason not to give the user the ability to remove styles they don't want for whatever reason, especially since this doesn't pose any drawbacks for users who want the full theme.

Reply:
It definitely shouldn’t use the name `styles` since that’s a plausible package name that might end up in Nixpkgs in future and break the default here. Generally speaking if people are going to want only a subset of styles I think it makes more sense to split this into multiple outputs or multiple packages and offer a package that combines them all.